Revert "SELinux: use deletion-safe iterator to free list"

This reverts commit 98a4d82a78914c2496989d669ecae04dedcf361f.

Bug: 22846070
Change-Id: Ic047824591e8d94f937c052846ab387864c04fd7
Signed-off-by: Jeff Vander Stoep <jeffv@google.com>

diff --git a/security/selinux/avc.c b/security/selinux/avc.c
index f3dbbc0..5c8e7cf 100644
--- a/security/selinux/avc.c
+++ b/security/selinux/avc.c
@@ -298,15 +298,13 @@
 
 static void avc_operation_free(struct avc_operation_node *ops_node)
 {
-	struct avc_operation_decision_node *od_node, *tmp;
+	struct avc_operation_decision_node *od_node;
 
 	if (!ops_node)
 		return;
 
-	list_for_each_entry_safe(od_node, tmp, &ops_node->od_head, od_list) {
-		list_del(&od_node->od_list);
+	list_for_each_entry(od_node, &ops_node->od_head, od_list)
 		avc_operation_decision_free(od_node);
-	}
 	kmem_cache_free(avc_operation_node_cachep, ops_node);
 }
 

